### ISSUE TITLE
Renaming .size() to .capacity() on ci::ConcurrentCircularBuffer<T>

### ISSUE THREAD

**Issue by num3ric** [cinder/Cinder#723]
https://github.com/cinder/Cinder/blob/master/include/cinder/ConcurrentCircularBuffer.h#L110

To follow the standards, and because at least twice I've made the mistake of trying to using .size() instead of say .isNotFull(), assuming it was dynamic. *(And yes I realize it's a stupid mistake to make given that doing a comparison would not be atomic.)*

Thoughts?

**Comment by richardeakin** [comment]
Makes sense to me. Also, should probably be `getCapacity()`, to indicate that it isn't some typedef'ed standard container.

Does it work to change the size implementation to something like the following?:

```
size_t getSize() const
{
    std::lock_guard<std::mutex> lock( mMutex );
    return mNumUnread;
}
```

If we were to make these changes, now seems like a good time (break it good!).

**Comment by andrewfb** [comment]
Addressed in #852
